code/GridFieldLimitItems.php

```diff
@@ -85,6 +121,23 @@ public function setRemoveFromTop($removeFromTop) {
         return $this;
     }
 
+    /**
+     * allows you to enforce the set limit by removing options to add new items
+     */
+    public function enforceLimit()
+    {
+        // ensure we are removing the buttons if these are
+        $this->onAfterManipulate(function(GridField $grid, SS_List $list){
+            if ($list->count() == $this->getMaxItems()) {
+                // var_dump($grid->getConfig());
+                // die();
```

Comment:
Should probably tidy up and remove these lines :-)

Reply:
Done!